### PR TITLE
doc: remove references to CreateArtifact restriction, clarify where g…

### PR DESCRIPTION
**Pull request by danabens**
* clarify where graph artifacts are rendered in Studio
* remove references to recently lifted `CreateArtifact` restriction

https://github.com/aws/sagemaker-experiments/issues/124

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/smexperiments/tracker.py [hunk 1]
```diff
@@ -591,10 +581,7 @@ def log_confusion_matrix(
         output_artifact=True,
     ):
         """Log a confusion matrix artifact which will be displayed in
-        studio.  Requires sklearn.
-
-        Note that this method must be run from a SageMaker context such as studio or training job
```
Comment (jerrypeng7773) [inline src/smexperiments/tracker.py:596]: but those methods indeed can only be called within sagemaker env?

File: src/smexperiments/tracker.py [hunk 2]
```diff
@@ -539,10 +532,7 @@ def log_roc_curve(
         output_artifact=True,
     ):
         """Log a receiver operating characteristic (ROC curve) artifact which will be displayed in
-        studio.  Requires sklearn.
-
-        Note that this method must be run from a SageMaker context such as studio or training job
-        due to restrictions on the CreateArtifact API.
+        Studio's Pipeline UI (not the Trial Component UI).  Requires sklearn.
```
Comment (jerrypeng7773) [inline src/smexperiments/tracker.py:535]: Im a bit confused here by:

1) what if I don't use these functions for pipeline execution, say, just for a single training job.
2) users are using experiment SDK, mentioning pipeline without more context is a bit confusing tho.

File: src/smexperiments/tracker.py (hunk 1 above)
Comment (danabens) [reply src/smexperiments/tracker.py:596]: `CreateTrialComponent` and `CreateArtifact` limitations were recently lifted.

File: src/smexperiments/tracker.py (hunk 2 above)
Comment (danabens) [reply src/smexperiments/tracker.py:535]: 1. the association and artifact will be created but there will be no way to visualize the graph from within Studio (for the time being)
2. ya, user may not know about pipelines or how they are rendered in Studio. maybe a link to a pipeline's blog post would provide context.

File: src/smexperiments/tracker.py (hunk 1 above)
Comment (jerrypeng7773) [reply src/smexperiments/tracker.py:596]: got it

File: src/smexperiments/tracker.py (hunk 2 above)
Comment (jerrypeng7773) [reply src/smexperiments/tracker.py:535]: yea true ... probably better to give more details here, e.g "if ur TC is created by/within a pipeline execution context, you can view the table from the step (who generates the TC) view." .... and a link.

**Review by jerrypeng7773 — APPROVED** [review]

**Pull request merged**